validphys2/src/validphys/coredata.py

```diff
@@ -189,6 +189,40 @@ def get_np_fktable(self):
 
         return fktable
 
+    def determine_pdfs(self, pdf):
+        """Determine the PDF (or PDFs) that should be used to be convoluted with this fktable.
+        Uses the `convolution_types` key to decide the PDFs.
+        If `convolution_types` is not defined, it returns the pdf object.
+        """
+        if self.convolution_types is None:
+            if self.hadronic:
+                return [pdf, pdf]
+            return [pdf]
+
+        conv_pdfs = []
+        for convolution_type in self.convolution_types:
+
+            # Check the type of convolutions that the fktable is asking for and match it to the PDF
+            if convolution_type == "UnpolPDF":
+                if pdf.is_polarized:
+                    if pdf.unpolarized_bc is None:
+                        raise ValueError(
+                            "The FKTable asked for an unpolarized PDF but received only polarized PDFs"
+                        )
+
+                    conv_pdfs.append(pdf.unpolarized_bc.make_only_cv())
+                else:
+                    conv_pdfs.append(pdf)
+
+            elif convolution_type == "PolPDF":
+                if not pdf.is_polarized:
+                    raise ValueError(
+                        "The FKTable asked for a polarized PDF, but the PDF received is unpolarized"
```

Comment:
```suggestion
                        """The FKTable asked for a polarized PDF, but the PDF received cannot be understood as polarized.
                        When using a polarized PDF make sure to include a boundary condition `unpolarized_bc: <pdf name>` whenever needed (`t0`, `dataspecs`...)."
```

Reply:
Updated the docs the reflect the changes here. Please have a look if you are fine with the additional modifications.